src/python/pants/backend/awslambda/python/rules_test.py

```diff
@@ -180,6 +183,15 @@ def handler(event, context):
     if sys.platform == "darwin":
         assert "`python_awslambda` targets built on macOS may fail to build." in caplog.text
 
+    global _first_run_of_deprecated_warning
```

Comment:
This is a big code smell, but also we don't typically test deprecation messages, as they are comprehensively tested generically. Manually observing that the right thing happens is sufficient.